arch/x86/x86_features.c

```diff
@@ -68,55 +57,55 @@ static inline uint64_t xgetbv(unsigned int xcr) {
 #endif
 }
 
-void Z_INTERNAL x86_check_features(void) {
+void Z_INTERNAL x86_check_features(struct x86_cpu_features *features) {
     unsigned eax, ebx, ecx, edx;
     unsigned maxbasic;
 
     cpuid(0, &maxbasic, &ebx, &ecx, &edx);
     cpuid(1 /*CPU_PROCINFO_AND_FEATUREBITS*/, &eax, &ebx, &ecx, &edx);
 
-    x86_cpu_has_sse2 = edx & 0x4000000;
-    x86_cpu_has_ssse3 = ecx & 0x200;
-    x86_cpu_has_sse41 = ecx & 0x80000;
-    x86_cpu_has_sse42 = ecx & 0x100000;
-    x86_cpu_has_pclmulqdq = ecx & 0x2;
+    features->has_sse2 = edx & 0x4000000;
+    features->has_ssse3 = ecx & 0x200;
+    features->has_sse41 = ecx & 0x80000;
+    features->has_sse42 = ecx & 0x100000;
+    features->has_pclmulqdq = ecx & 0x2;
 
     if (ecx & 0x08000000) {
         uint64_t xfeature = xgetbv(0);
 
-        x86_cpu_has_os_save_ymm = ((xfeature & 0x06) == 0x06);
-        x86_cpu_has_os_save_zmm = ((xfeature & 0xe6) == 0xe6);
+        features->has_os_save_ymm = ((xfeature & 0x06) == 0x06);
+        features->has_os_save_zmm = ((xfeature & 0xe6) == 0xe6);
     } else {
-        x86_cpu_has_os_save_ymm = 0;
-        x86_cpu_has_os_save_zmm = 0;
+        features->has_os_save_ymm = 0;
+        features->has_os_save_zmm = 0;
     }
 
     if (maxbasic >= 7) {
         cpuidex(7, 0, &eax, &ebx, &ecx, &edx);
 
         // check BMI1 bit
         // Reference: https://software.intel.com/sites/default/files/article/405250/how-to-detect-new-instruction-support-in-the-4th-generation-intel-core-processor-family.pdf
-        x86_cpu_has_vpclmulqdq = ecx & 0x400;
+        features->has_vpclmulqdq = ecx & 0x400;
 
         // check AVX2 bit if the OS supports saving YMM registers
-        if (x86_cpu_has_os_save_ymm) {
-            x86_cpu_has_avx2 = ebx & 0x20;
+        if (features->has_os_save_ymm) {
+            features->has_avx2 = ebx & 0x20;
         } else {
-            x86_cpu_has_avx2 = 0;
+            features->has_avx2 = 0;
```

Comment:
In `cpu_features.c`, there is a `memset`, so setting these to zero is no longer necessary.